lib/Cmts.ml

```diff
@@ -87,69 +118,45 @@ end = struct
-    of the first location is lower than that of the second location. *)
-let is_adjacent t (l1 : Location.t) (l2 : Location.t) =
-  Option.value_map (Source.string_between t.source l1.loc_end l2.loc_start)
-    ~default:false ~f:(fun btw ->
-      match String.strip btw with
-      | "" -> true
-      | "|" ->
-          Source.begins_line t.source l1
-          && Position.column l1.loc_start < Position.column l2.loc_start
-      | _ -> false)
-
-(** Whether the symbol preceding location [loc] is an infix symbol or a
-    semicolon. If it is the case, comments attached to the following item
-    should be kept after the infix symbol. *)
-let infix_symbol_before t (loc : Location.t) =
-  let pos_cnum = loc.loc_end.pos_cnum - 1 in
-  let loc_end = {loc.loc_end with pos_cnum} in
-  match Source.position_before t.source loc_end with
-  | Some loc_start ->
-      if loc_start.pos_cnum < loc.loc_end.pos_cnum then
-        let str = Source.string_at t.source loc_start loc.loc_end in
-        String.equal str ";" || Ast.String_id.is_infix str
-      else false
-  | None -> false
 
-(** Heuristic to choose between placing a comment after the previous location
-    or before the next one. *)
-let partition_after_prev_or_before_next t ~prev cmts ~next =
-  match CmtSet.to_list cmts with
-  | {loc; _} :: _ as cmtl when is_adjacent t prev loc -> (
-    match
-      List.group cmtl ~break:(fun l1 l2 ->
-          not (is_adjacent t (Cmt.loc l1) (Cmt.loc l2)))
-    with
-    | [cmtl] when is_adjacent t (List.last_exn cmtl).loc next ->
-        let open Location in
-        let same_line_as_prev l =
-          prev.loc_end.pos_lnum = l.loc_start.pos_lnum
-        in
-        let same_line_as_next l =
-          next.loc_start.pos_lnum = l.loc_start.pos_lnum
-        in
-        let prev, next =
-          if not (same_line_as_prev next) then
-            let next, prev =
-              List.partition_tf cmtl ~f:(fun {Cmt.loc= l1; _} ->
-                  same_line_as_next l1
-                  || (same_line_as_prev l1 && infix_symbol_before t prev)
-                  || not (same_line_as_prev l1))
-            in
-            (prev, next)
-          else ([], cmtl)
-        in
-        (CmtSet.of_list prev, CmtSet.of_list next)
-    | after :: befores ->
-        (CmtSet.of_list after, CmtSet.of_list (List.concat befores))
-    | [] -> impossible "by parent match" )
-  | _ -> (CmtSet.empty, cmts)
+  let partition src ~prev ~next cmts =
+    match to_list cmts with
+    | Cmt.{loc; _} :: _ as cmtl when is_adjacent src prev loc -> (
+      match
+        List.group cmtl ~break:(fun l1 l2 ->
+            not (is_adjacent src (Cmt.loc l1) (Cmt.loc l2)))
+      with
+      | [cmtl] when is_adjacent src (List.last_exn cmtl).loc next ->
+          let open Location in
+          let same_line_as_prev l =
+            prev.loc_end.pos_lnum = l.loc_start.pos_lnum
+          in
+          let decide loc =
+            match
```

Comment:
I'm not too convinced by the pattern matching on differences. What do you think about matching against the boolean result (`match ... = ..., ... = ... with true, true -> ...`).

Reply:
I made it is this because I have future plans of matching on a difference of 1, as for now there are issues with things like:
```ocaml
let foo = foooooooooo
(* comment not fitting on the previous line *)

let bar = bar
```

as the comment would be wrongfully attached to bar instead of foo and thus being moved one line lower (right in the middle)

Reply:
ok, let's go with integers then!